src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -157,13 +159,28 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
                         this.studentExam = studentExam;
                         this.exam = studentExam.exam!;
                         this.initIndividualEndDates(this.exam.startDate!);
+
                         // only show the summary if the student was able to submit on time.
                         if (this.isOver() && this.studentExam.submitted) {
                             this.examParticipationService
                                 .loadStudentExamWithExercisesForSummary(this.exam.course!.id!, this.exam.id!)
                                 .subscribe((studentExamWithExercises: StudentExam) => (this.studentExam = studentExamWithExercises));
                         }
-                        this.loadingExam = false;
+
+                        // Directly start the exam when we continue from a failed save
+                        if (this.examParticipationService.lastSaveFailed(this.courseId, this.examId)) {
+                            this.examParticipationService
+                                .loadStudentExamWithExercisesForConductionFromLocalStorage(this.exam.course!.id!, this.exam.id!)
+                                .subscribe((localExam: StudentExam) => {
+                                    this.studentExam = localExam;
+                                    this.examParticipationService.saveStudentExamToLocalStorage(this.exam.course!.id!, this.exam.id!, localExam);
```

Comment:
why do you need to save the exam to localstorage after you have just retrieved it?